galaxy_ng/app/common/proxy_url.py

```diff
@@ -0,0 +1,74 @@
+"""
+This module is temporarily added to address the
+issue AAH-291.
+
+The change that this module is solving
+is going to be implemented on pulpcore
+and then this module and its references
+can be removed or replaces with pulpcore
+implementation.
+"""
+from typing import List, Tuple, Optional
+from urllib3.util import parse_url, Url
+
+
+def get_parsed_url(parsed: Url, auth_items: Optional[List[Optional[str]]]) -> str:
```

Comment:
Instead of relying on the order in the auth_items to determine username / password, could
usename/password be passed in as a dict? Might give some flexibility if other components get added to the proxy auth later.  Could also use a fstring for building 'auth'.
(Though, this is "temporary" and a "shim" so doesn't matter a ton...)

Reply:
I preferred to keep as a sequence so I can use the same solution for when it has only `user` or both `user` and `password`

```py
":".join(["bruno"])
```

works also for

```py
":".join(["bruno", "1234"])
```

I think by its RFC there is no more components on the auth fragment.